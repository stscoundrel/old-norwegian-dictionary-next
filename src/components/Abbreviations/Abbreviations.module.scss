.column {
  margin-bottom: 30px;
}

.wrap {
  display: flex;
}

.abbreviation {
  flex: 1;
  position: relative;
  padding-right: 20px;

  &::after {
    content: 'â†’';
    position: absolute;
    right: 10px;
    top: 0;
  }
}
.explanation {
  flex: 2;
}


@media only screen and (min-width: 768px) {
  .abbreviations {
    display: flex;
    margin-top: 50px;
  }

  .column {
    flex-basis: 50%;
    width: 50%;
  }

  .explanation {
    flex: 3;
  }
}