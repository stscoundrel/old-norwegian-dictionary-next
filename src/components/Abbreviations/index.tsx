import { Abbreviation } from 'lib/services/abbreviations'
import styles from './Abbreviations.module.scss'

interface AbbreviationProps{
  abbreviations: Abbreviation[]
}

export default function Abbreviations({ abbreviations }: AbbreviationProps) {
  return (
    <div className={styles.abbreviations}>
      {abbreviations.length > 0
        && <div className={styles.column}>
          <h4>Abbreviations used:</h4>
          {abbreviations.map(({ abbreviation, explanation }) => (
            <dl className={styles.wrap} key={abbreviation}>
              <dt className={styles.abbreviation}>
                <strong>{abbreviation}</strong>
              </dt>
              <dd className={styles.explanation}>{explanation}</dd>
            </dl>
          ))}
        </div>
      }
    </div>
  )
}
