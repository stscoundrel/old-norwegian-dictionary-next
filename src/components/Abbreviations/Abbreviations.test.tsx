import React from 'react'
import renderer from 'react-test-renderer'
import Abbreviations from './index'

describe('Abbreviations component', () => {
  const abbreviations = [
    {
      abbreviation: 'adj.',
      explanation: 'Adjectiv.',
    },
    {
      abbreviation: 'f.',
      explanation: 'Feminin.',
    },
  ]

  test('Matches the snapshot', () => {
    const tree = renderer.create(
      <Abbreviations abbreviations={abbreviations} />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
