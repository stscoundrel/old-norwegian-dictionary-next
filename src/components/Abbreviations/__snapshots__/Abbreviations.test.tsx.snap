// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Abbreviations component Matches the snapshot 1`] = `
<div
  className="abbreviations"
>
  <div
    className="column"
  >
    <h4>
      Abbreviations used:
    </h4>
    <dl
      className="wrap"
    >
      <dt
        className="abbreviation"
      >
        <strong>
          adj.
        </strong>
      </dt>
      <dd
        className="explanation"
      >
        Adjectiv.
      </dd>
    </dl>
    <dl
      className="wrap"
    >
      <dt
        className="abbreviation"
      >
        <strong>
          f.
        </strong>
      </dt>
      <dd
        className="explanation"
      >
        Feminin.
      </dd>
    </dl>
  </div>
</div>
`;
